### PR TITLE
feat(ui): add notification ribbon atom to ui

### PR DESCRIPTION
**Pull request by dryhten**
re #228

#### Background

This is only the Notification Ribbon atom from the referenced issue.

#### How has this been tested?

Locally on styleguide.

### CODE REVIEW

File: app/components/ui/atoms/NotificationRibbon.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+export default styled.div`
+  background: ${backgroundColor};
+  color: ${th('colorTextReverse')};
+  padding: ${th('space.2')};
+  font-family: ${th('fontHeading')};
+  font-size: 14px;
+  line-height: 24px;
+  text-align: center;
+  a {
```
Comment (dryhten) [inline app/components/ui/atoms/NotificationRibbon.js]: Couldn't find a way to style specific styled components inside other styled components. Suggestions welcomed.

File: app/components/ui/atoms/NotificationRibbon.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+```js
+<NotificationRibbon type="error">
+  This is an error notification ribbon
+</NotificationRibbon>
+```
+
+Anchors also have a special style inside a ribbon
+
+```js
+;<NotificationRibbon type="primary">
```
Comment (diversemix) [inline app/components/ui/atoms/NotificationRibbon.md]: Is the semicolon important?

Comment (dryhten) [reply app/components/ui/atoms/NotificationRibbon.md]: No, I forgot it in there from when I was trying to style `Action` elements inside `NotificationRibbon`.

Comment (dryhten) [reply app/components/ui/atoms/NotificationRibbon.md]: I just removed it.

File: app/components/ui/atoms/NotificationRibbon.js (hunk 1 above)
Comment (g-sam) [reply app/components/ui/atoms/NotificationRibbon.js]: I think these styles should be applied to the component passed in by *that* component.

File: app/components/ui/atoms/NotificationRibbon.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import styled from 'styled-components'
+import { th } from '@pubsweet/ui-toolkit'
+
+const backgroundColor = ({ type = 'primary', theme }) =>
+  ({
+    error: theme.colorError,
+    success: theme.colorSuccess,
+    primary: theme.colorPrimary,
+  }[type])
```
Comment (tamlyn) [inline app/components/ui/atoms/NotificationRibbon.js]: We now have this helper function in three places. Surely it's time to factor it out into a common place?

File: app/components/ui/atoms/NotificationRibbon.js (hunk 1 above)
Comment (tamlyn) [reply app/components/ui/atoms/NotificationRibbon.js]: On one level this style rule makes sense: we specify the text colour for this element and we want all other child elements to have the same text colour. But using a naked `a` tag means we have to replicate any styling and behaviour attached to `Link`. What's more resend would probably be implemented using a `button` (via `Action`).

As discussed I think the best option is to create a separate `RibbonAction` component that applies the necessary text colour by extending `Action`.

Comment (dryhten) [reply app/components/ui/atoms/NotificationRibbon.js]: It seems that it's possible to apply style to specific styled components inside other styled components[1]. The problem is that the component we want to restyle is not actually a styled component. I will continue this discussion on coko's mattermost. For now I guess it's better to do the Action restyling in a different component so it's easier to change later on.

[1] - https://www.styled-components.com/docs/advanced#referring-to-other-components

File: app/components/ui/atoms/NotificationRibbon.js (hunk 3 above)
Comment (dryhten) [reply app/components/ui/atoms/NotificationRibbon.js]: While this takes type as input (as opposed to validationStatus), I agree, there should be a function like `getColor` somewhere in pubsweet that accounts for every possibility (error, success, warning, primary etc).

**Comment by dryhten** [comment]
I should mention that because we are using Actions now, there is this fancy effect when you click on the link in the ribbon -  a white border will show up for a brief period of time.

**Pull request merged**